src/back/msl/writer.rs

```diff
@@ -1160,7 +1194,11 @@ impl<W: Write> Writer<W> {
                                     if j != 0 {
                                         write!(self.out, ",")?;
                                     }
-                                    write!(self.out, "{}.{}[{}]", tmp, name, j)?;
+                                    write!(
```

Comment:
wait, do we need this section at all? It's only there because in this code:
```rust
Foo foo = { a ,b , c};
```
This code doesn't work if `b` is an array... But if we have it wrapped inside a structure, then we should no longer need a workaround!

Reply:
It's not quite that simple unfortunately. In the quad-vert code at the moment we have:
```metal
struct gl_PerVertex {
    metal::float4 gl_Position;
    float gl_PointSize;
    type6 gl_ClipDistance;
    type6 gl_CullDistance;
};
struct type10 {
    metal::float2 member;
    metal::float4 gl_Position1;
    float gl_PointSize1;
    type6 gl_ClipDistance1;
};
...
struct main2Output {
    metal::float2 member [[user(loc0), center_perspective]];
    metal::float4 gl_Position1 [[position]];
    float gl_PointSize1 [[point_size]];
    float gl_ClipDistance1 [[clip_distance]] [1];
};
vertex main2Output main2(
  main2Input varyings [[stage_in]]
) {
    gl_PerVertex _ = {};
    ...
    const auto _tmp = type10 {v_uv, _.gl_Position, _.gl_PointSize, {_.gl_ClipDistance.inner[0]}};
    return main2Output { _tmp.member, _tmp.gl_Position1, _tmp.gl_PointSize1, {_tmp.gl_ClipDistance1.inner[0]} };
}
```
If we remove the hack from `put_initialization_component` we get:
```metal
const auto _tmp = type10 {v_uv, _.gl_Position, _.gl_PointSize, _.gl_ClipDistance};
```
and that compiles fine, but doing a similar thing to `put_return_value` results in:
```metal
return main2Output { _tmp.member, _tmp.gl_Position1, _tmp.gl_PointSize1, _tmp.gl_ClipDistance1 };
```
which fails to compile with:
```